src/cascade/executor/argument_parser.py

```diff
@@ -99,70 +103,135 @@ def exit(self, status=0, message=None):
+        root_handler.setLevel(level)
         logging.root.addHandler(root_handler)
         logging.root.setLevel(level)
 
+        code_log = BaseArgumentParser._logging_configure_root_log(args.code_log, level)
+        BaseArgumentParser._logging_configure_mathlog(args.mvid, args.epiviz_log)
+        BaseArgumentParser._logging_individual_modules(args.logmod, args.modlevel)
+        if code_log:
+            logging.getLogger("cascade.math").info(f"Code log is at {code_log}")
+
+    @staticmethod
+    def _logging_configure_root_log(code_log_dir, level):
+        user_code_dir = code_log_dir / getuser() / "cascade"
+        try:
+            user_code_dir_exists = user_code_dir.exists()
+        except (OSError, PermissionError) as uce:
+            logging.warning(f"Cannot read user code dir {user_code_dir}: {uce}")
+            return
+        if not user_code_dir_exists:
+            try:
+                user_code_dir.mkdir(parents=True, exist_ok=True)
+            except PermissionError as pe:
+                logging.warning(f"Could not write to {user_code_dir} {pe}. "
+                                f"Not making a log file for code log.")
+                return
+
+        fname = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+        try:
+            log_file_descriptor, code_log_path = tempfile.mkstemp(
+                suffix=".log", prefix=fname, dir=user_code_dir, text=True)
+        except (OSError, PermissionError) as ose:
+            logging.warning(f"Could not write to file in {user_code_dir} to make a log. {ose}")
+            return
+
+        os.close(log_file_descriptor)
+        try:
+            code_handler = logging.FileHandler(code_log_path)
+        except (OSError, PermissionError) as fhe:
+            logging.warning(f"Could not write to file {code_log_path} so no code log: {fhe}")
+            return
+        code_handler.setLevel(level)
+        fmt = "%(levelname)s %(asctime)s %(pathname)s:%(lineno)d: %(message)s"
+        datefmt = "%y-%m-%d %H:%M:%S"
+        code_handler.setFormatter(logging.Formatter(fmt, datefmt))
+        # The memory handler reduces the number of writes to disk.
+        # It will flush writes when it encounters an ERROR.
+        outer_handler = logging.handlers.MemoryHandler(capacity=128000,
+                                                       target=code_handler)
+        outer_handler.setLevel(level)
+        logging.root.addHandler(outer_handler)
+        return code_log_path
+
+    @staticmethod
+    def _logging_configure_mathlog(mvid, epiviz_log_dir):
         # If the script is started with a model version ID, then it's
         # probably being run under EpiViz and should make a math log.
-        if args.mvid is not None and epiviz_log_dir.exists():
-            mvid = str(args.mvid)
-            math_log_dir = epiviz_log_dir / mvid
-            try:
+        if mvid is None:
+            logging.warning(f"There is no mvid, so will not write a mathlog.")
```

Comment:
Why not?

Reply:
I was looking for a sign this was running under EpiViz, but, yeah, why not?
Oh, wait, yes. If it's not under EpiViz, then should we overwrite the log.log that the modelers use? Yeah, why not.

Reply:
Without the mvid, I have nowhere to send this information, and it already goes to stderr. The directory name uses the mvid.